src/app/account/invitation/[inviteId]/new-user-account-form.tsx

```diff
@@ -53,25 +71,30 @@ const SetupAccountForm: FC<InviteProps & { onComplete(): void }> = ({ inviteId,
     })
 
     const { mutate: createAccount, isPending: isCreating } = useMutation({
-        mutationFn: (form: FormValues) => onCreateAccountAction({ inviteId, form }),
+        mutationFn: (formData: FormValues) => onCreateAccountAction({ inviteId, email, form: formData }), // Pass email
         onError: handleMutationErrorsWithForm(form),
-        async onSuccess(_, vals) {
+        async onSuccess(clerkUserId, vals) {
+            // clerkUserId is returned by onCreateAccountAction
             if (!signIn) {
-                reportError('unable to signin')
+                reportError(new Error('SignIn object not available'), 'SignIn not available post account creation')
+                return
+            }
+            if (!clerkUserId) {
+                reportError(new Error('Clerk User ID not returned from account creation'), 'Clerk User ID missing')
                 return
```

Comment:
can the action throw an exception if it didn't get a clerkUserId and then that'd be handled by the onError of the useMutation

Reply:
The server action now throws an error itself if it doesn't get a user ID.

This simplifies the client code because the onSuccess handler is now only for true successes, and the onError handler automatically catches this failure.